src/pre_process/m_initial_condition.fpp

```diff
@@ -607,30 +814,81 @@ contains
 
         tr = vr(idx)
         ti = vi(idx)
-        do i = 0, nl - 1
+        do i = 0, 5*n - 5
             call cdiv(vr(i), vi(i), tr, ti, cr, ci)
             vnr(i) = cr
             vni(i) = ci
         end do
 
+        ! Reassign vectors
+        xbr = 0d0
+        xbi = 0d0
+        do i = 1, 5
+            if (i < 3) then
+                do k = 0, n - 1
+                    xbr((i - 1)*(n + 2) + k + 1) = vnr((i - 1)*n + k)
+                    xbi((i - 1)*(n + 2) + k + 1) = vni((i - 1)*n + k)
+                end do
+            elseif (i == 3) then
+                do k = 0, n - 5
+                    xbr((i - 1)*(n + 2) + 2 + k + 1) = vnr((i - 1)*n + k)
+                    xbi((i - 1)*(n + 2) + 2 + k + 1) = vni((i - 1)*n + k)
+                end do
+            else
+                do k = 0, n - 1
+                    xbr((i - 1)*(n + 2) + k + 1) = vnr((i - 1)*n - 4 + k)
+                    xbi((i - 1)*(n + 2) + k + 1) = vni((i - 1)*n - 4 + k)
+                end do
+            end if
+        end do
+
+        ! rho at BC
+        xbr(0*(n + 2) + 0) = xbr(0*(n + 2) + 1) + xbr(2*(n + 2) + 1)*rho_mean(1)*mach
```

Comment:
where does the `+ 2` come from?

Reply:
`+2` or `-2` is for the insertion or deletion of boundary rows at y_beg and y_end.

Reply:
this is a magic number as well

Reply:
I removed magic numbers. I believe there is no more magic number...